packages/api/src/index.ts

```diff
@@ -46,14 +46,16 @@ export type {
   OsdkObjectLinksObject,
   SingleLinkAccessor,
 } from "./definitions/LinkDefinitions.js";
-export type { DerivedProperty } from "./derivedProperties/DerivedProperty.js";
+export type {
+  DerivedProperty,
+  DerivedPropertyCreator,
```

Comment:
We have been trying to avoid making too many types on the root. Perhaps we should do this as `DervivedProperty.Creator`?